specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2022-10-31-preview/iotcentral.json

```diff
@@ -0,0 +1,9326 @@
+      ]
+    },
+    "DataExplorerV1Destination": {
+      "description": "The azure data explorer destination definition.",
+      "x-ms-discriminator-value": "dataexplorer@v1",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Destination"
+        }
+      ],
+      "type": "object",
+      "properties": {
+        "clusterUrl": {
+          "description": "The resource URI of the Data Explorer instance.",
+          "type": "string"
+        },
+        "database": {
+          "description": "Name Data Explorer database where data should be written.",
+          "type": "string"
+        },
+        "table": {
+          "description": "The table within the Data Explorer database that will receive the data.",
+          "type": "string"
+        },
+        "authorization": {
+          "$ref": "#/definitions/DataExplorerV1DestinationAuth"
+        }
+      },
+      "required": [
+        "clusterUrl",
+        "database",
+        "table",
+        "authorization"
+      ]
+    },
+    "DataExplorerV1DestinationAuth": {
+      "description": "The authentication definition for azure data explorer destination.",
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "The kind of authentication to use.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "type"
+      ],
+      "discriminator": "type"
+    },
+    "DataExplorerV1DestinationServicePrincipalAuth": {
+      "description": "The authentication definition with service principal for azure data explorer destination.",
+      "x-ms-discriminator-value": "servicePrincipal",
+      "allOf": [
+        {
+          "$ref": "#/definitions/DataExplorerV1DestinationAuth"
+        }
+      ],
+      "type": "object",
+      "properties": {
+        "clientId": {
+          "description": "Service Principal client ID.",
+          "type": "string"
+        },
+        "tenantId": {
+          "description": "Service Principal tenant ID.",
+          "type": "string"
+        },
+        "clientSecret": {
+          "description": "Service Principal client secret.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "clientId",
+        "tenantId",
+        "clientSecret"
+      ]
+    },
+    "DataExplorerV1DestinationSystemAssignedManagedIdentityAuth": {
+      "description": "The authentication definition with system assigned managed identity for azure data explorer destination.",
+      "x-ms-discriminator-value": "systemAssignedManagedIdentity",
+      "allOf": [
+        {
+          "$ref": "#/definitions/DataExplorerV1DestinationAuth"
+        }
+      ]
+    },
+    "DataExportError": {
+      "description": "The data export error definition.",
+      "type": "object",
+      "properties": {
+        "code": {
+          "description": "The code for the error that occurred.",
+          "type": "string",
+          "readOnly": true
+        },
+        "message": {
```

Comment:
Seemingly trying to match the standard Azure error object, 1) could you just use the standard error response, or 2) if not, I recommend making these properties required as with the standard error response.

Reply:
Hi @heaths , we can't update the format now as it went out GA versions already, I checked error details "code" "message" is required, do you mean include the other fields as required as well?